java/ql/lib/semmle/code/java/frameworks/apache/IOGenerated.qll

```diff
@@ -1,684 +1,828 @@
+        "org.apache.commons.io;IOUtils;true;copy;(Reader,Appendable,CharBuffer);;Argument[2];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copy;(Reader,OutputStream);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copy;(Reader,OutputStream,Charset);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copy;(Reader,OutputStream,String);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copy;(Reader,Writer);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copy;(URL,File);;Argument[0];generated:open-url",
+        "org.apache.commons.io;IOUtils;true;copy;(URL,File);;Argument[1];generated:create-file",
+        "org.apache.commons.io;IOUtils;true;copy;(URL,OutputStream);;Argument[0];generated:open-url",
+        "org.apache.commons.io;IOUtils;true;copyLarge;(Reader,Writer);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copyLarge;(Reader,Writer,char[]);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copyLarge;(Reader,Writer,char[]);;Argument[2];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copyLarge;(Reader,Writer,long,long);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copyLarge;(Reader,Writer,long,long,char[]);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copyLarge;(Reader,Writer,long,long,char[]);;Argument[4];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;read;(Reader,char[]);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;read;(Reader,char[]);;Argument[1];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;read;(Reader,char[],int,int);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;read;(Reader,char[],int,int);;Argument[1];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;readFully;(Reader,char[]);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;readFully;(Reader,char[]);;Argument[1];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;readFully;(Reader,char[],int,int);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;readFully;(Reader,char[],int,int);;Argument[1];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;skip;(Reader,long);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;skipFully;(Reader,long);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;toByteArray;(Reader);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;toByteArray;(Reader,Charset);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;toByteArray;(Reader,String);;Argument[0];generated:write-file",
```

Comment:
These don't look like `write-file` sinks, do they? Apparently they only read from a file, not write into it.

Reply:
It turned out that these summaries were derived from the following hand written summaries.
"java.io;Writer;true;append;;;Argument[0];write-file"
"java.io;Writer;true;write;;;Argument[0];write-file"

Writer is an abstract class for writing character streams - so they probably shouldn't be write-file sinks.